sw.js

```diff
@@ -8,12 +8,21 @@ const putInCache = async (request, response) => {
   await cache.put(request, response);
 };
 
-const cacheFirst = async ({ request, fallbackUrl }) => {
+const cacheFirst = async ({ request, preloadResponsePromise, fallbackUrl }) => {
   // First try to get the resource from the cache
   const responseFromCache = await caches.match(request);
   if (responseFromCache) {
     return responseFromCache;
   }
+
+  // Next try to use the preloaded response, if it's there
+  const preloadResponse = await preloadResponsePromise;
```

Comment:
If preload is not enabled the value will be `undefined`. I presume that awaiting on undefined just falls through?

If so, awesome.

Reply:
Yes! `undefined === await undefined`